crates/agentgateway/src/client/mod.rs

```diff
@@ -297,9 +312,146 @@ impl Connector {
 				let rw = agent_hbone::RWStream {
 					stream: upgraded,
 					buf: Default::default(),
+					drain_tx: None,
 				};
 				Socket::from_hbone(Arc::new(stream::Extension::new()), pool_key.dst, rw)
 			},
+			Transport::DoubleHbone {
+				gateway_address,
+				gateway_identity,
+				waypoint_identity,
+				inner_tls,
+			} => {
+				if inner_tls.is_some() {
+					return Err(crate::http::Error::new(anyhow::anyhow!(
+						"todo: inner TLS after double hbone is not currently supported"
+					)));
+				}
+
+				tracing::debug!(
+					"will use DOUBLE HBONE: gateway {} -> workload {}",
+					gateway_address,
+					ep
+				);
+
+				// Create outer HBONE connection to network gateway
+				// The outer HBONE CONNECT request uses the service hostname (target) as the authority
+				// This tells the gateway what service we want to reach
+				let outer_uri = Uri::builder()
+					.scheme(Scheme::HTTPS)
+					.authority(match &target {
+						Target::Hostname(host, port) => format!("{}:{}", host, port),
+						Target::Address(addr) => addr.to_string(),
+					})
+					.path_and_query("/")
+					.build()
+					.expect("uri build should not fail");
+				let outer_req = ::http::Request::builder()
+					.uri(outer_uri)
+					.method(hyper::Method::CONNECT)
+					.version(hyper::Version::HTTP_2)
+					.body(())
+					.expect("builder with known status code should not fail");
+
+				// Connect to the network gateway at its HBONE port
+				let outer_pool_key = Box::new(WorkloadKey {
+					dst_id: vec![gateway_identity.clone()],
+					dst: gateway_address,
+				});
+				let mut pool = self
+					.hbone_pool
+					.clone()
+					.ok_or_else(|| crate::http::Error::new(anyhow::anyhow!("hbone pool disabled")))?;
+
+				let outer_upgraded = Box::pin(pool.send_request_pooled(&outer_pool_key, outer_req))
+					.await
+					.map_err(crate::http::Error::new)?;
+
+				// Wrap upgraded to implement tokio's Async{Write,Read}
+				let outer_rw = agent_hbone::RWStream {
+					stream: outer_upgraded,
+					buf: Default::default(),
+					drain_tx: None,
```

Comment:
Why is `drain_tx: None` set here? Do we not need to drain for outer_rw?

Reply:
The `outer_rw` doesn't need a drain_tx because it's a pooled connection (L366), and the pool itself manages draining and cleanup.